src/components/digit/Digit.tsx

```diff
@@ -0,0 +1,50 @@
+import Button from "../button/Button";
+import { FC } from "react";
+import { OperationTypeEnum } from "../operation/Opertaiton.enum.ts";
+import { ButtonVariantEnum } from "../button/Button.enum.ts";
+
+interface Props {
+  setNumber: (number: { left: string; right: string }) => void;
+  number: { left: string; right: string };
+  operator: { render: string; type: OperationTypeEnum };
+}
+const Digit: FC<Props> = ({
+  setNumber,
+  number: { left, right },
+  operator: { type },
+}) => {
+  const handleOnClickDigit = (digit: string) => {
+    if (
+      (left.length >= 3 && OperationTypeEnum.INIT === type) ||
+      right.length >= 3
+    )
+      return alert("숫자는 3자리까지만 입력 가능합니다.");
+
+    setNumber(
+      Object.assign(
+        { left, right },
+          OperationTypeEnum.INIT === type
+          ? { left: `${Number(`${left}${digit}`)}` }
+          : { right: `${Number(`${right}${digit}`)}` },
+      ),
+    );
```

Comment:
리액트 useState의 상태 업데이트는 비동기적으로 처리되기 때문에 이전 상태를 참조해야하는 경우 setState 함수 내에서 참조하셔야 예상되지 않은 동작을 피할 수 있어요.
```suggestion
    setNumber((prev) => {
      const isUpdatingLeftOperand = operationType === OperationTypeEnum.INIT;
      if (isUpdatingLeftOperand) {
        // 왼쪽 피연산자 갱신
        return { ...prev, left: `${Number(`${prev.left}${digit}`)}` };
      } else {
        // 오른쪽 피연산자 갱신
        return { ...prev, right: `${Number(`${prev.right}${digit}`)}` };
      }
    });
```